Add the "no data" option for the Metric Threshold rule (#4967)
elastic/observability-docs: docs/en/observability/metrics-threshold-alert.asciidoc

In the *Advanced options*, you can change the number of consecutive runs that must meet the rule conditions before an alert occurs.
The default value is `1`.

[discrete]

> [[if-there-is-no-data]]

Reviewer: @alaudazzi wdyt about adding a screenshot here of "If there is no data" section?

Author: I've added the screenshot. Not sure about the actual added value, though.

Reviewer 2: Agreed, Ari. I don't think there's any value added here. Let's remove this.

== If there is no data

In the **If there is no data**, you can control how alerts behave when data stops being reported by selecting one of the following options:

* **Recover active alerts**: Recover active alerts when data is missing; no new alerts are created.
* **Alert me about the missing data**

>    ** **with groupBy**: Trigger a “no data” alert when a previously detected group stops reporting data; not recommended for dynamically scaling infrastructures that start and stop nodes automatically.

Reviewer: ```suggestion
   ** **When using "Group alerts by"**: Trigger a “no data” alert when a previously detected group stops reporting data; not recommended for dynamically scaling infrastructures that start and stop nodes automatically.
```

Author: We can just keep the name of the option in bold as it appears on the UI, to stay consistent with the rest of the bulleted list.

Author: Fixed

>    ** **without groupBy**: Trigger a “no data” alert when no data is returned during rule execution or when the rule fails to query {{es}}.

Reviewer: ```suggestion
   ** **When "Group alerts by" is not used**: Trigger a “no data” alert when no data is returned during rule execution or when the rule fails to query {{es}}.
```

Author: Same comment above.

Author: Fixed

In the *Advanced options*, you can change the number of consecutive runs that must meet the rule conditions before an alert occurs.
The default value is `1`.

[discrete]
[[if-there-is-no-data]]
== If there is no data

> In the **If there is no data**, you can control how alerts behave when data stops being reported.

Reviewer 2: Is the word `panel` or `dialog` missing here? For example:

In the **If there is no data** _panel_, you can control...

Or maybe `section`?

Author: I thought about the same while editing...it's not a dialog, might be a panel, it's an area of the window. We can check with UX how this is called.
